R/Quosure.R

```diff
@@ -0,0 +1,307 @@
+#' Reproducible class with environment and code.
+#'
+#' @slot code (`character`) to reproduce the environment
+#' @slot env (`environment`) environment which content was generated by the evaluation
+#' of the `code` slot.
+setClass(
+  "Quosure",
+  representation(code = "character", env = "environment"),
+  prototype(code = character(0), env = new.env(parent = parent.env(.GlobalEnv)))
+)
+
+#' Initialize `Quosure` object
+#'
+#' Initialize `Quosure` object with `code` and `env`. In order to have `Quosure` reproducible
+#' one needs to initialize with `env` which can be reproduced by the `code`. Alternatively, one
+#' can create an empty `Quosure` and evaluate the expressions in this object.
+#' @name new_quosure
+#'
+#' @param code (`character(1)` or `language`) code to evaluate. Also acceptps and stores comments
+#' @param env (`environment`) name of the code element.
+#'
+#' @export
+setGeneric("new_quosure", function(code = character(0), env = parent.env(.GlobalEnv)) {
+  standardGeneric("new_quosure")
+})
+
+#' @rdname new_quosure
+#' @export
+setMethod(
+  "new_quosure", signature(code = "missing", env = "missing"),
+  function(code, env) {
+    methods::new("Quosure")
+  }
+)
+
+#' @rdname new_quosure
+#' @export
+setMethod(
+  "new_quosure", signature(code = "character", env = "environment"),
+  function(code = character(0), env = parent.env(.GlobalEnv)) {
+    checkmate::check_class(env, "environment")
+    checkmate::check_string(code)
```

Comment:
I would say that these checks are obsolete as the arg checks are done by method dispatch

UPDATE:
maybe leave string check because it includes length validation